test/lib/ufe/testAttribute.py

```diff
@@ -342,3 +405,148 @@ def testAttributeDouble3(self):
 
         # Then make sure that new UFE value matches what it in USD.
         self.assertVectorAlmostEqual(ufeAttr.get(), usdAttr.Get())
+
+        self.runUndoRedo(ufeAttr,
+                         ufe.Vector3d(vec.x()-1.0, vec.y()-2.0, vec.z()-3.0))
+
+    def testObservation(self):
```

Comment:
Add observation testing.  Asserting on the number of notifications received is how I found the duplicate USD attribute Set() notification problem.